plugins/inventory/proxmox.py

```diff
@@ -0,0 +1,350 @@
+                    # /facts are returned as dict in 'data'
+                    data = json['data']
+                    break
+                else:
+                    # /hosts 's 'results' is a list of all hosts, returned is paginated
+                    data = data + json['data']
+                    break
+
+            self._cache[self.cache_key][url] = data
+
+        return self._cache[self.cache_key][url]
+
+    def _get_nodes(self):
+        return self._get_json("%s/api2/json/nodes" % self.proxmox_url)
+
+    def _get_pools(self):
+        return self._get_json("%s/api2/json/pools" % self.proxmox_url)
+
+    def _get_lxc_per_node(self, node):
+        return self._get_json("%s/api2/json/nodes/%s/lxc" % (self.proxmox_url, node))
+
+    def _get_qemu_per_node(self, node):
+        return self._get_json("%s/api2/json/nodes/%s/qemu" % (self.proxmox_url, node))
+
+    def _get_members_per_pool(self, pool):
+        ret = self._get_json("%s/api2/json/pools/%s" % (self.proxmox_url, pool))
+        return ret['members']
+
+    def _get_node_ip(self, node):
+        ret = self._get_json("%s/api2/json/nodes/%s/network" % (self.proxmox_url, node))
+
+        for iface in ret:
+            try:
+                return iface['address']
+            except Exception:
+                return None
+
+    def _get_vm_config(self, node, vmid, vmtype, name):
+        ret = self._get_json("%s/api2/json/nodes/%s/%s/%s/config" % (self.proxmox_url, node, vmtype, vmid))
+
+        vmid_key = 'vmid'
+        vmid_key = self.to_safe('%s%s' % (self.get_option('facts_prefix'), vmid_key.lower()))
+        self.inventory.set_variable(name, vmid_key, vmid)
+
+        vmtype_key = 'vmtype'
+        vmtype_key = self.to_safe('%s%s' % (self.get_option('facts_prefix'), vmtype_key.lower()))
+        self.inventory.set_variable(name, vmtype_key, vmtype)
+
+        for config in ret:
+            key = config
+            key = self.to_safe('%s%s' % (self.get_option('facts_prefix'), key.lower()))
+            value = ret[config]
+            try:
+                # fixup disk images as they have no key
+                if config == 'rootfs' or config.startswith(('virtio', 'sata', 'ide', 'scsi')):
+                    value = ('disk_image=' + value)
+
+                if isinstance(value, int) or ',' not in value:
+                    value = value
+                # split off strings with commas to a dict
+                else:
+                    # skip over any keys that cannot be processed
+                    try:
+                        value = dict(key.split("=") for key in value.split(","))
+                    except Exception:
+                        continue
+
+                self.inventory.set_variable(name, key, value)
+            except NameError:
+                return None
+
+    def _get_vm_status(self, node, vmid, vmtype, name):
+        ret = self._get_json("%s/api2/json/nodes/%s/%s/%s/status/current" % (self.proxmox_url, node, vmtype, vmid))
```

Comment:
Shouldn't this call be only made when `self.get_option('want_facts')` is `True`?

Reply:
I'm not sure, the information for this call is only used (when get_facts == False) to put all detected VMs in groups depending on their current running state.

On the other hand, it basically is a fact.

I'll switch it around :-)